SpatialGDK/Source/SpatialGDKTests/Examples/SpatialGDKTestGuidelines.h

```diff
@@ -0,0 +1,56 @@
+/*
+0. How to run tests:
+	- Tests can be run via Session Frontend in UE4 Editor
+
+	- Tests can be run via command line
+
+1. Folder structure
+	- Tests are located in a separate SpatialGDKTests module.
+	So for every component that needs to be tested - it has to be exposed via corresponding macro.
+	E.g. `class FRPCContainer` -> `class SPATIALGDK_API FRPCContainer`, to make FRPCContainer testable.
+	WThe macro is different in each module SPATIALGDK_API, SPATIALGDKSERVICES_API etc.
+
+	- The folder structure inside SpatialGDKTests should resemble the folder structure for components being tested.
+	E.g. If file tested is
+	`UnrealGDK\SpatialGDK\Source\SpatialGDK\Public\Utils\RPCContainer.cpp`,
+	then the corresponding test should be located in
+	`UnrealGDK\SpatialGDK\Source\SpatialGDKTests\SpatialGDK\Utils\RPCContainer\RPCContainerTest.cpp`.
+	There should be a folder with the same name as component tested. It should include the test and all the supporting files.
+
+2. Test definitions (check TestDefinitions.h for more info)
+    - We have defined 3 types of Macro to be used when writing tests:
+	(https://docs.unrealengine.com/en-US/Programming/Automation/TechnicalGuide/index.html has more information)
+	TEST - the most common simple test
+	COMPLEX_TEST - same as simple test, but allows having multiple test cases run through the same test function body.
+	DEFINE_LATENT_COMMAND... - used to run tests that require more than 1 thread.
+
+	- There are 5 types of mock objects we can use in tests:
```

Comment:
Is this just common terminology?

Reply:
It is, as far as I know. Of course, we're free to define our own terminology, however, I find the 5 types there sufficient so far.

Reply:
I mean is this just enumerating the things. It's not saying these are things we have helpers for or anything?

Reply:
We don't have helpers, it's just an enumeration of mock types.

Reply:
In which case what is the purpose of it. Is it basically to keep the terminology of the team consistent or is it because we want to add more stuff in the future.

Reply:
Yes, it is there to keep the terminology consistent.